editor/plugins/script_editor_plugin.cpp

```diff
@@ -2319,10 +2320,30 @@ void ScriptEditor::_update_script_names() {
 	}
 
 	Vector<_ScriptEditorItemData> sedata_filtered;
-	for (int i = 0; i < sedata.size(); i++) {
-		String filter = filter_scripts->get_text();
-		if (filter.is_empty() || filter.is_subsequence_ofn(sedata[i].name)) {
-			sedata_filtered.push_back(sedata[i]);
+
+	String filter = filter_scripts->get_text();
+
+	if (filter.is_empty()) {
+		sedata_filtered = sedata;
+	} else {
+		HashMap<String, int> name_to_index;
+		PackedStringArray search_names;
+
+		for (int i = 0; i < sedata.size(); i++) {
+			name_to_index[sedata[i].name] = i;
+			search_names.append(sedata[i].name);
+		}
+
+		Vector<FuzzySearchResult> results;
+		FuzzySearch fuzzy;
+		fuzzy.set_query(filter);
+		fuzzy.case_sensitive = false;
+		fuzzy.search_all(search_names, results);
+
+		for (const FuzzySearchResult &res : results) {
```

Comment:
It could be good to add a new public field `FuzzySearchResult::original_index` to avoid needing hashmaps for post processing. Originally I wanted to template it for arbitrary types of targets, for ease of associating data with the results, but it seemed overkill at that point.

Reply:
would you like me to make a pr for that or do you want to do that as part of these prs?

Reply:
If you have the time to do a PR for that it would be great, you'd probably make a cleaner implementation that fits with what's already there. I can mark these two as Drafts and then update them to use the new functionality when yours is merged.